### ISSUE TITLE
Metric input shape

### ISSUE THREAD

**Issue by Johanmkr** [SFI-Visual-Intelligence/Collaborative-Coding-Exam#68]
We need to decide on the shape of `y_pred` and `y_true` that goes into the `forward` function of each metric class. 

This is also discussed in #54

**Comment by hzavadil98** [comment]
y_pred -  NxC, y_true - Nx1. The whole main is set up that way. #52

**Comment by Seilmast** [comment]
As mentioned by @hzavadil98 and in #52, probably use the raw vectors/matrices in the train validation loop
That is 
y_true with the shape Nx1
y_pred with the shape NxC
Where N is the batchsize and C is the number of classes. 
Y_pred is also the raw output of your model. No argmax or softmax before passing it to the metric classes.
